hail/python/hail/utils/misc.py

```diff
@@ -96,7 +96,7 @@ def range_table(n, n_partitions=None) -> 'hail.Table':
     -------
     :class:`.Table`
     """
-    check_positive_and_in_range('range_table', 'n', n)
+    check_nonnegative_and_in_range('range_table', 'n', n)
```

Comment:
Do we want the same thing for MatrixTables?

Reply:
Yes: https://github.com/hail-is/hail/pull/7376